libraries/AP_GPS/AP_GPS_PX4.cpp

```diff
@@ -39,9 +39,9 @@ AP_GPS_PX4::AP_GPS_PX4(AP_GPS &_gps, AP_GPS::GPS_State &_state, AP_HAL::UARTDriv
 }
 
 
-#define MS_PER_WEEK             7*24*3600*1000LL
-#define DELTA_POSIX_GPS_EPOCH   315964800LL*1000LL
-#define LEAP_MS_GPS_2014        16*1000LL
+const uint64_t MS_PER_WEEK            = ((uint64_t)7)*24*3600*1000;
+const uint64_t DELTA_POSIX_GPS_EPOCH  = ((uint64_t)3657)*24*3600*1000;
+const uint64_t LEAP_MS_GPS_2016       = ((uint64_t)17)*1000;
```

Comment:
It's not mission critical/holding this up, for example the u-blox driver will report a time before it necessarily has all of the conversion data to UTC and we lock that as the starting time, (and we don't even ensure we are starting from GPS as the base time).

I was mostly just observing that this will need someone to remember and come touch it every time the number of leap seconds change.

How does the zubax get the conversion? IIRC it could take upto 12.5 minutes to have the correct GPS->UTC conversion data. Does it just shift its reported time (and APM does accept that, there is some code in to make sure that the system clock doesn't jump after the first GPS fix's time).